alibi/exceptions.py

```diff
@@ -27,3 +27,14 @@ class AlibiPredictorReturnTypeError(AlibiException):
     an unexpected or unsupported type.
     """
     pass
+
+
+class NotFittedError(AlibiException):
+    """
+    This exception is raised whenever a compulsory call to a `fit` method has not been carried out.
+    """
+
+    def __init__(self, object_name: str):
+        super().__init__(
+            f"This {object_name} instance is not fitted yet. Call 'fit' with appropriate arguments first."
+        )
```

Comment:
I've decided to put the error message here and allow it to be parametrized with the object name. The message is generic enough to apply to any object that as a compulsory `fit` method so I don't think it's necessary to define messages by hand in the code (which will inevitably become different from each other).